src/main/java/net/snowflake/ingest/streaming/internal/DataValidationUtil.java

```
@@ -846,14 +842,9 @@ private static String sanitizeValueForExceptionMessage(Object value) {
    * UTF-16 surrogate, for example.
    */
   private static void verifyValidUtf8(String input, String columnName, String dataType) {
-    CharsetEncoder charsetEncoder =
-        StandardCharsets.UTF_8
-            .newEncoder()
-            .onMalformedInput(CodingErrorAction.REPORT)
-            .onUnmappableCharacter(CodingErrorAction.REPORT);
-    try {
-      charsetEncoder.encode(CharBuffer.wrap(input));
-    } catch (CharacterCodingException e) {
+    String roundTripStr =
+        new String(input.getBytes(StandardCharsets.UTF_8), StandardCharsets.UTF_8);
+    if (!input.equals(roundTripStr)) {
       throw valueFormatNotAllowedException(columnName, input, dataType, "Invalid Unicode string");
```

Should we throw an exception with `roundTripStr` as well?